api/client/diff.go

```diff
@@ -19,11 +19,8 @@ import (
 func (cli *DockerCli) CmdDiff(args ...string) error {
 	cmd := cli.Subcmd("diff", []string{"CONTAINER"}, "Inspect changes on a container's filesystem", true)
 	cmd.Require(flag.Exact, 1)
-	cmd.ParseFlags(args, true)
 
-	if cmd.Arg(0) == "" {
```

Comment:
w/o this check we now see:
```
docker diff ""
Error response from daemon: 404 page not found
```
instead of a nice error message. I'd prefer to keep the old code/error msg.